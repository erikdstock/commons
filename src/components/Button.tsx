import styled from "@emotion/styled"
import { withStyleProps } from "../util/styleComposition"

export const Button = withStyleProps({
  name: "Button",
  component: "button",
  defaultVariant: "primary",
})
