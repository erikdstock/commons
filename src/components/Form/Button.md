React component example:

```js
<Button>Push Me</Button>
```

```js
<div style={{width: "500px"}}>
  <Button block >Block Button</Button>
</div>
```

```js
  <Button outline>Outline Button</Button>
```


To render an example as highlighted source code add a `static` modifier:

```jsx static
import React from 'react';
```

Examples with all other languages are rendered only as highlighted source code, not an actual component:

```html
<Button size="large">Push Me</Button>
```

Any [Markdown](http://daringfireball.net/projects/markdown/) is **allowed** _here_.
