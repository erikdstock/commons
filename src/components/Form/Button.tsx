import React from "react"
import { Button as ReButton, ButtonProps } from "rebass"
// TODO: Block button, figure out how to document props

export const Button: React.FunctionComponent<ButtonProps> = props => (
  <ReButton
    variant="primary"
    disabled={props.disabled || props.variant === "disabled"}
    {...props}
  />
)
