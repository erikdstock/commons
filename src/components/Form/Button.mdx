---
Button
---
import { Button } from './Button'
import { Box, Flex } from '../../'
import { Playground } from 'docz'

# Buttons

Standard Rebass Buttons.

### Primary:

<Playground>
  <Flex justifyContent="space-around">
    <Button onClick={() => alert("ðŸŒ¹")}>
      Normie
    </Button>
  </Flex>
</Playground>


### Variants:
<Playground >
  <Flex justifyContent="space-around">
  <Box p={3}>
    <Button variant="outline" onClick={() => alert("ðŸŒˆ")}>Outline</Button>
  </Box>
  <Box p={3}>
    <Button variant="inactive" onClick={() => alert("ðŸ¥€")}>Inactive</Button>
  </Box>
  </Flex>
</Playground>

# Cool



