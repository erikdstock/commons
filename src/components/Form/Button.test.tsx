import React from 'react'
import {Button} from './Button'
import renderer from 'react-test-renderer'

describe("Button", () => {
  const tree = renderer.create(<Button>Click me</Button>).toJSON()
  it("renders correctly", () => {
    expect(tree).toMatchSnapshot()
  })
  // it("is red by default", () => {
  //   expect(tree).toHaveStyleRule('background', colors.red)
  // })
  // it('is not 100% width by default', () => {
  //   expect(tree).toHaveStyleRule('width', undefined)
  // })
  // it("is white if `outline`", () => {
  //   expect(renderer.create(<Button outline>Click me</Button>).toJSON())
  //     .toHaveStyleRule('background', colors.white)
  // })
  // it("is 100% wide if `block`", () => {
  //   expect(renderer.create(<Button block>Click me</Button>).toJSON())
  //     .toHaveStyleRule('width', "100%")
  // })
})