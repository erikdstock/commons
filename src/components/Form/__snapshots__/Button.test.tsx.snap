// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  font-size: inherit;
  color: white;
  background-color: blue;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: bold;
  border: 0;
  border-radius: 4px;
  border-radius: 4px;
}

<button
  className="sc-bdVaJa c0"
  color="white"
  disabled={false}
  fontSize="inherit"
  fontWeight="bold"
>
  Click me
</button>
`;
